### PR TITLE
feat(CL): add bank sends to positions

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #XXX

## What is the purpose of the change

As part of an offline conversation, we agreed to remove the BankSend logic from [this PR](https://github.com/osmosis-labs/osmosis/pull/3497) and create a separate PR for it. This PR introduces the BankSend logic for creating and leaving a position.

### CODE REVIEW

File: x/concentrated-liquidity/lp.go [hunk 1]
```diff
@@ -147,3 +164,19 @@ func (k Keeper) updatePosition(ctx sdk.Context, poolId uint64, owner sdk.AccAddr
 	// The returned amounts are rounded down to avoid returning more to clients than they actually deposited.
 	return actualAmount0.TruncateInt(), actualAmount1.TruncateInt(), nil
 }
+
+func (k Keeper) sendCoinsBetweenPoolAndUser(ctx sdk.Context, denom0, denom1 string, amount0, amount1 sdk.Int, sender, receiver sdk.AccAddress) error {
```
Comment (p0mvn) [inline x/concentrated-liquidity/lp.go:168]: godoc and tests please

**Review by p0mvn — COMMENTED** [review]
Nice work! Looks great to me once godoc and tests added!

File: x/concentrated-liquidity/lp_test.go [hunk 2]
```diff
@@ -276,3 +278,108 @@ func mergeConfigs(dst *lpTest, overwrite *lpTest) {
...
+		"only asset0 is greater than sender has, position creation (user to pool)": {
+			coin0:       sdk.NewCoin("eth", sdk.NewInt(100000000000000)),
+			coin1:       sdk.NewCoin("usdc", sdk.NewInt(1000000)),
+			expectError: true,
+		},
+		// "only asset1 is greater than sender has, position creation (user to pool)": {
+		// 	coin0:       sdk.NewCoin("eth", sdk.NewInt(1000000)),
+		// 	coin1:       sdk.NewCoin("usdc", sdk.NewInt(100000000000000)),
+		// 	expectError: true,
+		// },
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/lp_test.go:311]: for some reason, everything works as expected but this...

if the second asset is too large, we end up BankSending the coin0 but not coin1...

Comment (czarcas7ic) [reply x/concentrated-liquidity/lp_test.go:311]: Actually, I suppose this makes sense. While this test case will be odd and show a partial send, it wouldn't actually get committed and would get reverted due to the error right? Yeah I guess for error cases I should just not check the resulting amounts

**Review by p0mvn — APPROVED** [review]
Great work! I'm good with merging this now (before merge to `main` is complete)

**Pull request merged**